src/Juvix/Backends/Michelson/Compilation/VirtualStack.hs

```diff
@@ -41,22 +41,44 @@ data T lamType
       }
   deriving (Show, Eq)
 
+--------------------------------------------------------------------------------
+-- Usage Information
+--------------------------------------------------------------------------------
+
+type Saved = Bool
```

Comment:
Hmm can we `newtype` this?

Reply:
I'd rather not, that way I can use it in boolean expressions.

```haskell
isUsageFree :: Usage -> Bool
isUsageFree (Usage x saved)
  | x == mempty
      || x == one && saved =
    False
  | otherwise = True
``

for example

Reply:
Ah, OK